packages/bus/src/Brokers/RMQEventBus.ts

```diff
@@ -0,0 +1,395 @@
+    }
+
+    if (!this.container.isRegistered(type)) {
+      throw new EventHandlerNotFound(type.name);
+    }
+    const handlerInstance = this.container.resolve(type);
+
+    const eventNames: string[] = this.reflectEventsNames(type);
+    if (eventNames?.length === 0) {
+      throw new NoSubscriptionsFound(handlerInstance);
+    }
+
+    for (const eventName of eventNames) {
+      const eventHandlers = this.handlers.get(eventName);
+      if (!eventHandlers) {
+        continue;
+      }
+
+      const handlerIndex = eventHandlers.indexOf(handlerInstance);
+      eventHandlers.splice(handlerIndex, 1);
+
+      await this.channel?.unbindQueue(
+        this.configuration.clientQueue,
+        this.configuration.exchange,
+        eventName
+      );
+    }
+  }
+
+  public async publish<T>(event: Event<T>): Promise<void> {
+    const { type, payload, correlationId, createdAt } = event;
+    if (!this.channel) {
+      throw new IllegalOperation(this);
+    }
+
+    await this.backOffRetry.acquire(() => {
+      this.channel?.publish(
+        this.configuration.exchange,
+        type,
+        Buffer.from(JSON.stringify(payload)),
+        {
+          contentType: 'application/json',
+          mandatory: true,
+          persistent: true,
+          replyTo: this.REPLY_QUEUE_NAME,
+          correlationId,
+          timestamp: createdAt.getTime()
+        }
+      );
+    });
+  }
+
+  public async execute<T, R>(command: Command<T, R>): Promise<R> {
+    const { type, payload, correlationId, createdAt, ttl, expectReply } =
+      command;
+
+    if (!this.channel) {
+      throw new IllegalOperation(this);
+    }
+
+    const waiter = expectReply
+      ? this.getWaiter(ttl, correlationId)
+      : Promise.resolve([]);
+
+    try {
+      this.channel.sendToQueue(
+        this.APP_QUEUE_NAME,
+        Buffer.from(JSON.stringify(payload)),
+        {
+          type,
+          correlationId,
+          timestamp: createdAt.getTime(),
+          replyTo: this.REPLY_QUEUE_NAME
+        }
+      );
+
+      const [response]: [R] = (await waiter) as [R];
+
+      return response as R;
+    } finally {
+      this.subject.removeAllListeners(correlationId);
+    }
+  }
+
+  public async destroy(): Promise<void> {
+    if (!this.client) {
+      return;
+    }
+
+    if (this.channel) {
+      await this.channel.waitForConfirms();
+      await Promise.all(
+        this.consumerTags.map(consumerTag => this.channel?.cancel(consumerTag))
+      );
+      await this.channel.close();
+    }
+
+    await this.client.close();
+    this.clear();
+  }
+
+  private getWaiter<R>(ttl: number, correlationId: string): Promise<[R]> {
+    return Promise.race([
+      once(this.subject, correlationId) as Promise<[R]>,
+      new Promise<never>((_, reject) =>
+        setTimeout(reject, ttl, new NoResponse(ttl))
+      )
+    ]);
+  }
+
+  private reflectEventsNames<T, R>(
+    handler: EventHandlerConstructor<T, R>
+  ): string[] {
```

Comment:
I don't see any reason to use generics here

```suggestion
  private reflectEventsNames(
    handler: EventHandlerConstructor<unknown, unknown>
  ): string[] {
```